client/dns_client.ml

```diff
@@ -26,6 +26,14 @@ module Pure = struct
     end, { protocol ; query ; key = record_type }
 
   (* name: the originally requested domain name. *)
+  (* NOTE that this function compresses answers:
+     foo.example CNAME 500 bar.example
+     bar.example A 300 1.2.3.4
+     is compressed to:
+     foo.example A 300 1.2.3.4
+     -> which is fine for applications (i think so)
+     -> which is struggling for the cache (not entirely sure about this tbh)
+     -> it is not clear whether it meets the DNS specifications nicely *)
```

Comment:
after re-thinking, I guess for "good answers" nobody cares. but for "no such domain" / "no data" answers (former is "this domain name does not exist", latter "this resource record type does not exist for this domain name -- see https://tools.ietf.org/html/rfc2308 for error handling), this may make a difference since the client gets "the wrong SOA" -- i.e. an unexpected authority (where the requested domain name does not match) - which strict clients may want to drop on the floor instead of accepting it.. I have no concrete example we can test it on, but I'm sure we can find some on the internet. :)

Reply:
I also wonder how this plays in with DNSSEC - if I request `www.foo.example` I would intuitively expect there to be a signed path leading to the answer of `www.bar.example has A record 1.2.3.4` (signed by bar.example?), in that case including the CNAME might make sense?